### ISSUE TITLE
Athena timestamp query format not supported

### ISSUE THREAD

**Issue by avikkumar1983** [grafana/athena-datasource#131]
For such S3 bucket prefix:

```
s3://bucket/prefix/2021/01/01/00/
s3://bucket/prefix/2021/01/01/01/
s3://bucket/prefix/2021/01/01/02/
```

Athena supports querying date as below:
```
SELECT *
FROM my_ingested_data
WHERE datehour >= '2021/11/09/00' 
```

But when I write query in Grafana it always try to convert as below:

WHERE
$__timeFilter(datehour)

it converts to 
`datehour BETWEEN TIMESTAMP '2022-02-18 19:19:33' AND TIMESTAMP '2022-02-25 19:19:33'`

In this way it fails querying the data. Also even if it able to query it will never use partition as the partition is not on timestamp (rather its a divided hour, day format). Is there a way to achieve the query in grafana for athena data source?

**Comment by andresmgot** [comment]
hi @avikkumar1983,

the `$__timeFilter` macro allows you to specify the format so you could try something like `$__timeFilter(datehour, 'yyyy/MM/dd/HH')` but that would still only work if the field is a timestamp I believe. Another option is to use the macro `$__dateFilter` but that would only work for `date` fields.

If none of the above fits your use case, then we would need to adapt or create a new [macro](https://github.com/grafana/athena-datasource#macros) for that.

**Comment by avikkumar1983** [comment]
Thanks for the reply. This one worked but internally it converted the query like below:
`parse_datetime(datehour,'yyyy/MM/dd/HH') BETWEEN TIMESTAMP '2022-03-10 23:48:26' AND TIMESTAMP '2022-03-11 00:48:26'`

This caused a high response time because it's basically using a column function & not using any partition index anymore. But in Athena its using right partition & running pretty fast.

**Comment by andresmgot** [comment]
Thanks for the details!

I am not sure I fully understand your last paragraph but something we can do is adding more macros to support that use case. For example, we could have a macro `$__timeFromRaw('yyyy/MM/dd/HH')` that returns the initial value of the time range as you need it: `2022/03/10/23`.

To add a new macro, you need to add it here in the backend to be processed: 

https://github.com/grafana/athena-datasource/blob/main/pkg/athena/macros.go

And here in the fronted so it's listed as a suggestion in the code editor:

https://github.com/grafana/athena-datasource/blob/main/src/Suggestions.ts

In case you want to give it a try!

**Comment by avikkumar1983** [comment]
Thanks for the details. I am hosting my grafana on an ec2 instance. Do I just need to add these to datasource code directory of grafaba/athena in ec2 instance?

**Comment by andresmgot** [comment]
no problem @avikkumar1983, I have added this to our backlog. We will let you know if we get to it, in the meantime someone from the community can give this a try!

**Comment by kevinwcyu** [comment]
Hi @avikkumar1983. Two new macros have been added: `$__rawTimeFrom` and `$rawTimeTo`. They're available in v1.0.4. Please see https://github.com/grafana/athena-datasource#macros to see their usage.
